### DISCUSSION TITLE
Stack Failed in AWS - Hubs Cloud

### DISCUSSION THREAD

**Discussion by jesudarrigrande** [mozilla/hubs#3158] [Hubs Cloud Questions] [answered]
Hi! I want to create my own event with Hubs Cloud and I followed the step by step video of "How to deploy Hubs Cloud AWS", I created my 2 domains with Route 53, create the Key pair and did everithing that the tutorial said from the beginning, but the creation process of my stack failed. Please help me, this is for my title project to graduate from university. 

(I appreciate any help, if someone speaks Spanish even better)

<img width="1280" alt="Screen Shot 2020-10-12 at 13 28 23" src="https://user-images.githubusercontent.com/42739894/95769957-0d6c7380-0c8f-11eb-945c-691bd91befda.png">
<img width="1278" alt="Screen Shot 2020-10-12 at 13 29 01" src="https://user-images.githubusercontent.com/42739894/95769983-18bf9f00-0c8f-11eb-9b0a-bc7ce514eca7.png">
<img width="1277" alt="Screen Shot 2020-10-12 at 13 29 20" src="https://user-images.githubusercontent.com/42739894/95770001-207f4380-0c8f-11eb-8783-6e84377a2e7d.png">
<img width="1280" alt="Screen Shot 2020-10-12 at 13 29 40" src="https://user-images.githubusercontent.com/42739894/95770017-2412ca80-0c8f-11eb-967e-6b966c5935b0.png">

**Comment by robinkwilson** [comment, marked as answer]
Did you agree to the Terms + Conditions and hit "Subscribe"? It looks like you didn't "Subscribe" to hubs cloud yet so it failed. I've never seen this before, so I'm curious if you didn't go through the aws menus all the way (before the cloudformation template).

**Reply by jesudarrigrande** [reply]
Yes! I did just like the tutorial said. I agree the terms and conditions, hit suscribe and then the page redirect me to "create stack", where it was the cloudformation template information. I complete all the steps, but it failed in the last one. What can I do? Can I delete the actual stack and do the process again using the same domains and key pair? Is there any charge for doing the process again?

**Reply by robinkwilson** [reply]
Absolutely you can delete then remake the stack! No charge likely extremely minimal (AWS bills hourly for running instances, so you're good redeploying). 

I'd go through the whole process again just in case. It's a complex process, so sometimes it doesn't work the first time.

**Reply by jesudarrigrande** [reply]
Ok!! Would you suggest that I also buy another 2 domains on route 53? Or I can do the whole process again with the domains I already have?

**Reply by robinkwilson** [reply]
Same domains! Same keys.

**Reply by jesudarrigrande** [reply]
Great, thank you!!! I am going to try that

**Reply by jesudarrigrande** [reply]
I deleted the stack and create a new one with the same domains and keys, following the instructions of the AWS Troubleshooting, were I selected the "Disabled" option for "Rollback on failure". This is the error the stack failed on, what does it mean?

<img width="1280" alt="Screen Shot 2020-10-13 at 13 36 43" src="https://user-images.githubusercontent.com/42739894/95889725-29d4e280-0d59-11eb-88e5-aff0496791c3.png">

On another hand, the domain support send me this email to get my website listed on Google, Yahoo, Bing and all the other main search engines. Is this related to the stack problem?

<img width="860" alt="Screen Shot 2020-10-13 at 13 38 53" src="https://user-images.githubusercontent.com/42739894/95890103-b384b000-0d59-11eb-9d80-dfab49eed19a.png">

**Reply by robinkwilson** [reply]
Interesting, this doesn't tell me the classification of errors or any error message -- could be a problem on our side if it's not descriptive -- but I'm not sure that's the first error event. That one looks like the _last error_ event.

**Is that the first error message scrolling down as far as you can? Down past the _first_ red error should be green and blue.** Usually it tells us which step failed so we can debug it.
For Cloudformation deploys, the **_last_** error is at the top. The **_first_** is at the bottom. 


If that's not the case and that **is** the first error message, please email **hubs@mozilla.com** and mention this discussion thread.

**Reply by jesudarrigrande** [reply]
Oh ok!! I understand, sorry the one that I sent was the last error. This is the first one: (down past it is all green and blue as you said)

<img width="1279" alt="Screen Shot 2020-10-14 at 16 42 53" src="https://user-images.githubusercontent.com/42739894/96037689-8c52df00-0e3c-11eb-91e7-33eb7d4805b3.png">

**Reply by robinkwilson** [reply]
Perfect, that's what we're looking for!!

**Seems like your AWS account is missing some permissions to create resources/access AWSLambdaInternal service.** Are you using an account with that permission available or a managed AWS account for a larger organization where your permissions to create resources is limited?

I can't help you get that AWS permission since it's a problem on the AWS account permissions side. **Ask your AWS account manager to give you creation permissions for AWSLambdaInternal resources.**

**Alternatively, the fastest way to get you up and running is to create a new AWS account** (then you have ALL AWS create permissions available on that account) and go through the Hubs Cloud Cloudformation template again.

**Reply by robinkwilson** [reply]
[Temporarily marked comment as answered] Unmark this comment as "answered", when you add more information.
